lib/httpadapter/fetchadapter.js

```diff
@@ -40,13 +40,24 @@ class FetchAdapter {
           options
         );
 
-        return res.json();
+        const contentType = res.headers.get('content-type');
+        if (contentType && contentType.indexOf('application/json') !== -1) {
+          return res.json();
+        } else {
+          throw new HttpError(await res.text(), {
+            code: res.statusCode
+          });
+        }
```

Comment:
Code will behave the same, but I would remove the `else` as follows:

```js
if (contentType && contentType.indexOf('application/json') !== -1) {
  return res.json();
}

throw new HttpError(await res.text(), {
  code: res.statusCode
});
```

Reply:
Fair enough, I usually try to avoid return short-circuit, I don't know why I did here. I'll invert the logic to leave the return at the end of the block and short-circuit when a trowing error

Reply:
done